cloud/google/machineactuator.go

```diff
@@ -266,10 +266,21 @@ func (gce *GCEClient) Create(cluster *clusterv1.Cluster, machine *clusterv1.Mach
 		if gce.machineClient == nil {
 			labels[BootstrapLabelKey] = "true"
 		}
+		serviceAccounts := []*compute.ServiceAccount{nil}
+		if util.IsMaster(machine) {
```

Comment:
I assuming that the service account is needed for the machine controller. The assumption here is that the machine controller only runs on the master. Could we call out that assumption in a comment.

Reply:
It's actually needed for kubernetes' cloud provider specific code. K8s needs to be about to read compute API for faster node deletion, create advanced routes for pod CIDRs on nodes,  etc...